app/scripts/proactive/model/Task.js

```diff
@@ -381,11 +397,110 @@ define(
                 ]
 
                 this.controlFlow = {};
-                this.on("change", function (eventName, event) {
-                    undoManager.save()
+
+                // get the GI of the Task
+                var genericInformation = this.get["Generic Info"];
+
+                this.on("change", function(eventName, error) {
+                  if (eventName) {
+                    if (eventName._changing) {
+                      // Check if Generic Info doc has been changed and generate new link if needed
+                      if (eventName.changed.hasOwnProperty('Generic Info') && eventName.changed["Generic Info"]["Property Value"] != "") {
+
+                        var genericInformation = eventName.changed["Generic Info"];
+                        var hasDocumentation;
+
+                        // Regenerate new documentation url
+                        for (var i in genericInformation) {
+                          if (genericInformation[i]["Property Name"].toLowerCase() === 'task.documentation') {
+                            hasDocumentation = true;
+                            var fileContent = "Documentation for the Job \"" + this.get('Name') + "\" \n" + "\n" + "\n";
+                            var fileContent = fileContent + "Documentation value: " + genericInformation[i]["Property Value"] + "\n";

```

Comment:
I think it would be better with a variable that contains the value "Property Value"